zyte_test_websites/articles/extraction.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, cast
+
+from clear_html import clean_node, cleaned_node_to_html
+from web_poet import field
+from zyte_common_items import (
+    Article,
+    ArticleNavigationPage,
+    ArticlePage,
+    Author,
+    ProbabilityRequest,
+    Request,
+)
+
+if TYPE_CHECKING:
+    from parsel import Selector
+
+
+__all__ = [
+    "TestArticleNavigationPage",
+    "TestArticlePage",
+]
+
+
+def article_body_html_processor(value: Selector, page: Any) -> str:
+    """Extract and clean the HTML from the provided selector."""
+    cleaned_node = clean_node(value.root, page.url)
+    return cleaned_node_to_html(cleaned_node)
+
+
+class TestArticlePage(ArticlePage):
+    def validate_input(self) -> Article | None:
+        if self.css(".article-content"):
+            return None
+        return cast("Article", self.no_item_found())
```

Comment:
hm, I wonder why is cast needed; I was hoping our types will be properly inferred (https://github.com/zytedata/zyte-common-items/blob/74c4f51cca05cfe001185c6cadd4aeea76731dbe/zyte_common_items/pages/base.py#L26)